### PR TITLE
[New Rule] Clearing Windows Console History

### LINKED ISSUE #1622: [New Rule] Clearing Windows Console History
## Description

Identifies when an user attempts to clear console history. An adversary may clear the command history of a compromised account to conceal the actions undertaken during an intrusion.

### Target indexes

"winlogbeat-*", "logs-endpoint.events.*", "logs-windows.*"

### Target Operating Systems

Windows

### Query

```
process where event.type in ("process_started", "start") and
   (process.name : ("pwsh.exe", "powershell.exe", "powershell_ise.exe") or process.pe.original_file_name in
   ("pwsh.exe", "powershell.exe", "powershell_ise.exe") and
   process.args : ("Set-PSReadlineOption","Remove-Item","*ConsoleHost*history*","Clear-History","-HistorySaveStyle","SaveNothing")
```

### References

- https://community.sophos.com/sophos-labs/b/blog/posts/powershell-command-history-forensics

### PR DESCRIPTION
**Pull request by austinsonger**
## Issues

Resolves #1622 

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/windows/defense_evasion_clearing_windows_console_history.toml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+rule_id = "b5877334-677f-4fb9-86d5-a9721274223b"
+severity = "low"
+tags = ["Elastic", "Host", "Windows", "Threat Detection", "Defense Evasion"]
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+process where event.type in ("process_started", "start") and
+   (process.name : ("pwsh.exe", "powershell.exe", "powershell_ise.exe") or process.pe.original_file_name in
+     ("pwsh.exe", "powershell.exe", "powershell_ise.exe") and
```
Comment (w0rk3r) [inline rules/windows/defense_evasion_clearing_windows_console_history.toml:32]:
```suggestion
     ("pwsh.dll", "powershell.exe", "powershell_ise.exe") and
```
As we saw last time, the original name for pwsh.exe is pwsh.dll

File: rules/windows/defense_evasion_clearing_windows_console_history.toml [hunk 2]
```diff
@@ -0,0 +1,52 @@
+[metadata]
+creation_date = "2021/11/22"
+maturity = "production"
+updated_date = "2021/11/22"
+
+[rule]
+author = ["Austin Songer"]
+description = """
+Identifies when an user attempts to clear console history. An adversary may clear the command history of a compromised account to conceal
```
Comment (w0rk3r) [inline rules/windows/defense_evasion_clearing_windows_console_history.toml:9]:
```suggestion
Identifies when a user attempts to clear console history. An adversary may clear the command history of a compromised account to conceal
```

File: rules/windows/defense_evasion_clearing_windows_console_history.toml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+severity = "low"
+tags = ["Elastic", "Host", "Windows", "Threat Detection", "Defense Evasion"]
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+process where event.type in ("process_started", "start") and
+   (process.name : ("pwsh.exe", "powershell.exe", "powershell_ise.exe") or process.pe.original_file_name in
+     ("pwsh.exe", "powershell.exe", "powershell_ise.exe") and
+	 process.args : ("Set-PSReadlineOption","Remove-Item","*ConsoleHost*history*","Clear-History","-HistorySaveStyle","SaveNothing")
```
Comment (w0rk3r) [inline rules/windows/defense_evasion_clearing_windows_console_history.toml:33]: This need to be elaborated. Same problem as https://github.com/elastic/detection-rules/pull/1565#discussion_r755014400

**Review by w0rk3r — CHANGES_REQUESTED** [review]
Left some changes. Thanks

File: rules/windows/defense_evasion_clearing_windows_console_history.toml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+process where event.type in ("process_started", "start") and
+   (process.name : ("pwsh.exe", "powershell.exe", "powershell_ise.exe") or process.pe.original_file_name in
+     ("pwsh.dll", "powershell.exe", "powershell_ise.exe") and
+	 process.args : ("Clear-History", "Remove-Item", "*ConsoleHost*history*") or
+	 process.args : ("Remove-Item","(Get-PSReadlineOption).HistorySavePath") or
+	 process.args : ("Set-PSReadlineOption","-HistorySaveStyle","SaveNothing"))
```
Comment (w0rk3r) [inline rules/windows/defense_evasion_clearing_windows_console_history.toml:35]: This still an OR condition, for example. this query would match any command line that contains `Remove-Item` alone. You need an AND condition here.

See [this](https://github.com/elastic/detection-rules/pull/1565#discussion_r755959162) comment on another PR, you can adjust this one based on that query

**Review by w0rk3r — CHANGES_REQUESTED** [review]

File: rules/windows/defense_evasion_clearing_windows_console_history.toml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+type = "eql"
+
+query = '''
+process where event.type in ("process_started", "start") and
+   (process.name : ("pwsh.exe", "powershell.exe", "powershell_ise.exe") or process.pe.original_file_name in
+     ("pwsh.dll", "powershell.exe", "powershell_ise.exe") and
+	 process.args : "Clear-History" or
+	 process.args : "Remove-Item" and process.args : "*ConsoleHost*history*" or
+	 process.args : "Remove-Item" and process.args : "*Get-PSReadlineOption*.HistorySavePath" or
+	 process.args : "Set-PSReadlineOption" and process.args : ("-HistorySaveStyle","SaveNothing")
```
Comment (w0rk3r) [inline rules/windows/defense_evasion_clearing_windows_console_history.toml:36]:
```suggestion
process where event.action == "start" and
  (process.name : ("powershell.exe", "pwsh.exe", "powershell_ise.exe") or process.pe.original_file_name == "PowerShell.EXE") and
     (process.args : "*Clear-History*" or
     (process.args : ("*Remove-Item*", "rm") and process.args : ("*ConsoleHost_history.txt*", "*(Get-PSReadlineOption).HistorySavePath*")) or
     (process.args : "*Set-PSReadlineOption*" and process.args : "*SaveNothing*"))
```
Some adjusts

**Review by w0rk3r — APPROVED** [review]
Just some additional changes on the query, then LGTM

**Review by Samirbous — APPROVED** [review]
👍  maybe bump up the risk score to Medium.

**Pull request merged**